### PR TITLE
Fix(Bug): Automatic_function_calling config persists across requests, causing UNEXPECTED_TOOL_CALL when tools=None

### LINKED ISSUE #1818: automatic_function_calling config persists across requests, causing UNEXPECTED_TOOL_CALL when tools=None
### Summary
The `automatic_function_calling` configuration in `GenerateContentConfig` persists across API requests within the same client session. When a request with tools enabled (AFC active) is followed by a request with `tools=None`, Gemini still attempts function calls, resulting in `FinishReason.UNEXPECTED_TOOL_CALL` with malformed responses (`candidate.content = None`).

### Steps to Reproduce

1. Create a Gemini client with a model that supports function calling
2. Make a request with tools enabled and `automatic_function_calling` active
3. Make a subsequent request with `tools=None` (no tools declared)
4. Observe `FinishReason.UNEXPECTED_TOOL_CALL` with empty response content

### Expected Behavior
When `tools=None` or empty, AFC should be disabled automatically for that request, regardless of previous requests' configurations.

### Actual Behavior
AFC remains enabled from previous requests. Gemini attempts function calls from conversation context even when no tools are declared, resulting in:
- `FinishReason.UNEXPECTED_TOOL_CALL`
- `response.candidates[0].content = None`
- Log message: `"INFO:google_genai.models:AFC is enabled with max remote calls: 10"`

### Environment
- SDK: `google-genai` (Python)
- Integration: CrewAI (multi-agent framework making multiple LLM calls with varying tool configurations)
- Model: `gemini-2.5-pro`

### Workaround
Explicitly set `config.automatic_function_calling = None` when `config.tools` is None/empty:

```python
if config and hasattr(config, 'tools') and not config.tools:
    if hasattr(config, 'automatic_function_calling'):
        config.automatic_function_calling = None
```

### Root Cause Analysis
AFC appears to be a client-level or session-level setting that doesn't reset between individual `generate_content()` calls. This is problematic for frameworks that make multiple LLM calls with different tool configurations in the same session.

### Related Issues
- LangChain.js issue with UNEXPECTED_TOOL_CALL: https://github.com/langchain-ai/langchainjs/issues/8589

### PR DESCRIPTION
**Pull request by Muhammad-Rebaal**
Fixes #1818 

Hi @janasangeetha !

I've identified and fixed this issue
The bug was in the `should_disable_afc()`  function in `google/genai/_extra_utils.py`. The function defaulted to enabling AFC even when tools=None or tools=[], because it checked the `automatic_function_calling` config before checking whether any tools were actually provided.

Thank you!

### CODE REVIEW

**Comment by janasangeetha** [comment]
Hey @Muhammad-Rebaal 
Thanks for your contributions!
The branch is out-of-date. Please update the same.

**Review by BenjaminKazemi — COMMENTED** [review]
Please test `tests/afc/test_get_max_remote_calls_for_afc.py` as well. Thanks for your contribution.

File: google/genai/tests/afc/test_should_disable_afc.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def test_afc_enable_unset_max_1():
               ),
           )
       )
-      is False
```
[the remark this answers is not in the document]
Comment (BenjaminKazemi) [reply google/genai/tests/afc/test_should_disable_afc.py]: Thanks! In addition to that would you mind adding an example code that reproduce the bug? AFC logic may seem simple in the first glance but there are corner cases. Would be idea if you could add an example to regenerate the issue or refer me to an existing one.
